### ISSUE TITLE
cm-type added to CMW array / record

### ISSUE THREAD

**Issue by nedmsmith** [thomas-fossati/draft-ftbs-rats-msg-wrap#24]
There has been a discussion thread related to adding cm-type to the CMW array / record as a way to indicate which RATS conceptual messages are behind the wrapper. The following CDDL is suggested:

```
cmw = [ type, value, cm-bits ]
cm-bits = uint .bits cm-type
cm-type = &(
                reference-values: 0
                endorsements: 1
                evidence: 2
                attestation-results: 3
}
```
The justification for cm-bits is that a conceptual message structure could contain more than one type of conceptual message. For example a corim could contain conditional endorsed values given a stateful environment-map. Additionally, it could contain trust anchor stores, or attestation results and all of the conceptual messages that were used as Verifier inputs.

**Comment by thomas-fossati** [comment]
Are trust anchor stores another type of conceptual messages?

And more generally: do we envisage the set of conceptual messages to be extensible?

Because that'd also require specifying the procedures for adding new bits to `cm-type`.

**Comment by nedmsmith** [comment]
> Are trust anchor stores another type of conceptual messages?
> 
> And more generally: do we envisage the set of conceptual messages to be extensible?
> 
> Because that'd also require specifying the procedures for adding new bits to `cm-type`.

It is a philosophical question IMO. Trust anchors could be Endorsements if a vendor wishes to delegate well-known TA too a less well-known TA. 
TAs could be a form of appraisal policy where appraisal policies are also conceptual messages (though out of scope under the current charter), they could be added in the future. So, yes conceptual messages are extensible IMO.
Yes, it makes sense to describe procedures for adding new bits to cm-type. Presumably, it adds an entry to the cm-type enumeration list. Bits not in the enumeration list but included in the uint .bits probably should be ignored unless a profile dictates otherwise.

**Comment by thomas-fossati** [comment]
> It is a philosophical question IMO. Trust anchors could be Endorsements if a vendor wishes to delegate well-known TA too a less well-known TA. TAs could be a form of appraisal policy where appraisal policies are also conceptual messages (though out of scope under the current charter), they could be added in the future. So, yes conceptual messages are extensible IMO. Yes, it makes sense to describe procedures for adding new bits to cm-type. Presumably, it adds an entry to the cm-type enumeration list. Bits not in the enumeration list but included in the uint .bits probably should be ignored unless a profile dictates otherwise.

OK, I'll add something to that effect to #20

**Comment by thomas-fossati** [comment]
This issue has been addressed in #20 

The remaining question about extensibility:

> [...] it makes sense to describe procedures for adding new bits to cm-type. Presumably, it adds an entry to the cm-type enumeration list. Bits not in the enumeration list but included in the uint .bits probably should be ignored unless a profile dictates otherwise.

is tracked by #26
